framework/src/modules/interoperability/mainchain/commands/message_recovery.ts

```diff
@@ -0,0 +1,132 @@
+import { CCMsg, StoreCallback, MessageRecoveryParams } from '../../types';
+import { BaseInteroperabilityCommand } from '../../base_interoperability_command';
+import { MainchainInteroperabilityStore } from '../store';
+import { getIDAsKeyForStore, swapChainIDs } from '../../utils';
+import { BaseInteroperableAPI } from '../../base_interoperable_api';
+
+export class MessageRecoveryCommand extends BaseInteroperabilityCommand {
+	public id = COMMAND_ID_MESSAGE_RECOVERY;
+	public name = 'messageRecovery';
+	public schema = messageRecoveryParams;
+
+	// TODO
+	// eslint-disable-next-line @typescript-eslint/require-await
+	public async verify(): Promise<VerificationResult> {
+		throw new Error('Method not implemented.');
+	}
+
+	public async execute(context: CommandExecuteContext<MessageRecoveryParams>): Promise<void> {
+		const { transaction, params, getAPIContext, logger, networkIdentifier, getStore } = context;
+		const apiContext = getAPIContext();
+
+		let chainIdAsBuffer = getIDAsKeyForStore(params.chainID);
+
+		const updatedCCMs: Buffer[] = [];
+		const deserializedCCMs = params.crossChainMessages.map(serializedCCMsg =>
+			codec.decode<CCMsg>(ccmSchema, serializedCCMsg),
+		);
+		for (const ccm of deserializedCCMs) {
+			const apisWithBeforeRecoverCCM = [...this.interoperableCCAPIs.values()].filter(api =>
+				Reflect.has(api, 'beforeRecoverCCM'),
+			) as Pick<Required<BaseInteroperableAPI>, 'beforeRecoverCCM'>[];
+			for (const api of apisWithBeforeRecoverCCM) {
+				await api.beforeRecoverCCM({
+					ccm,
+					trsSender: transaction.senderAddress,
+					eventQueue: apiContext.eventQueue,
+					getAPIContext,
+					logger,
+					networkIdentifier,
+					getStore,
+					feeAddress: EMPTY_FEE_ADDRESS,
+				});
+			}
+
+			const recoveryCCM: CCMsg = {
+				...ccm,
+				fee: BigInt(0),
+				status: CCM_STATUS_RECOVERED,
+			};
+			const encodedUpdatedCCM = codec.encode(ccmSchema, recoveryCCM);
+			updatedCCMs.push(encodedUpdatedCCM);
+		}
+
+		const interoperabilityStore = this.getInteroperabilityStore(getStore);
+
+		const doesTerminatedOutboxAccountExist = await interoperabilityStore.terminatedOutboxAccountExist(
+			chainIdAsBuffer,
+		);
+
+		if (!doesTerminatedOutboxAccountExist) {
+			throw new Error('Terminated outbox account does not exist.');
+		}
+
+		const terminatedChainOutboxAccount = await interoperabilityStore.getTerminatedOutboxAccount(
+			chainIdAsBuffer,
+		);
+		const terminatedChainOutboxSize = terminatedChainOutboxAccount.outboxSize;
+
+		const proof = {
+			size: terminatedChainOutboxSize,
+			indexes: params.idxs,
+			siblingHashes: params.siblingHashes,
+		};
+
+		const hashedUpdatedCCMs = updatedCCMs.map(ccm => hash(ccm));
+
+		const outboxRoot = regularMerkleTree.calculateRootFromUpdateData(hashedUpdatedCCMs, proof);
+
+		await interoperabilityStore.setTerminatedOutboxAccount(chainIdAsBuffer, {
+			outboxRoot,
+		});
+
+		for (const ccm of deserializedCCMs) {
+			const newCcm = swapChainIDs(ccm);
```

Comment:
`newCCM`, ,maybe its better to keep the abbreviation in capital as much as we can

Reply:
Confusing thing was in some places we use `ccm` and in some places we use `CCMsg`. I think `CCMsg` is better because ending a word with a lowercase letter yields better camelCase naming. I tried to end the word with a lowercase letter due to that. What do you think about using `CCMsg` instead of `CCM` in this PR? Same problem exists with `ID`.